client/components/mma/upgrade/UpgradeSupportAmountForm.tsx

```diff
@@ -0,0 +1,226 @@
+import { benefitsConfiguration } from '../shared/benefits/BenefitsConfiguration';
+import { BenefitsSection } from '../shared/benefits/BenefitsSection';
+import type { UpgradeSupportInterface } from './UpgradeSupportContainer';
+import { UpgradeSupportContext } from './UpgradeSupportContainer';
+
+function validateChoice(
+	currentAmount: number,
+	chosenAmount: number | null,
+	minAmount: number,
+	maxAmount: number,
+	isOtherAmountSelected: boolean,
+	mainPlan: PaidSubscriptionPlan,
+): string | null {
+	const chosenOptionNum = Number(chosenAmount);
+	if (!chosenAmount && !isOtherAmountSelected) {
+		return 'Please make a selection';
+	} else if (chosenOptionNum === currentAmount) {
+		return 'You have selected the same amount as you currently contribute';
+	} else if (!chosenAmount || isNaN(chosenOptionNum)) {
+		return 'There is a problem with the amount you have selected, please make sure it is a valid amount';
+	} else if (!isNaN(chosenOptionNum) && chosenOptionNum < minAmount) {
+		return `There is a minimum ${
+			mainPlan.billingPeriod
+		}ly contribution amount of ${mainPlan.currency}${minAmount.toFixed(
+			2,
+		)} ${mainPlan.currencyISO}`;
+	} else if (!isNaN(chosenOptionNum) && chosenOptionNum > maxAmount) {
+		return `There is a maximum ${
+			mainPlan.billingPeriod
+		}ly contribution amount of ${mainPlan.currency}${maxAmount.toFixed(
+			2,
+		)} ${mainPlan.currencyISO}`;
+	}
+	return null;
+}
+
+function isAboveThreshold(chosenAmount: number | null): EmotionJSX.Element {
+	const chosenOptionNum = Number(chosenAmount);
+	if (!isNaN(chosenOptionNum) && chosenOptionNum < 10) {
+		return (
+			<BenefitsSection
+				benefits={benefitsConfiguration['contributions']}
+			/>
+		);
+	} else {
+		return (
+			<BenefitsSection
+				benefits={benefitsConfiguration['supporterplus']}
+			/>
+		);
+	}
+}
+
+export const UpgradeSupportAmountForm = () => {
+	const upgradeSupportContext = useContext(
+		UpgradeSupportContext,
+	) as UpgradeSupportInterface;
+
+	const priceConfig = (supporterPlusPriceConfigByCountryGroup[
+		upgradeSupportContext.mainPlan.currencyISO as CurrencyIso
+	] || supporterPlusPriceConfigByCountryGroup.international)[
+		calculateMonthlyOrAnnualFromBillingPeriod(
+			upgradeSupportContext.mainPlan.billingPeriod,
+		)
+	];
+
+	const amountLabel = (amount: number) => {
+		return `${upgradeSupportContext.mainPlan.currency}${amount} per ${upgradeSupportContext.mainPlan.billingPeriod}`;
+	};
+	const currentAmount = upgradeSupportContext.mainPlan.price / 100;
+
+	const mainPlan = upgradeSupportContext.mainPlan;
+
+	const otherAmountLabel = `Choose an amount (${upgradeSupportContext.mainPlan.currency} per ${upgradeSupportContext.mainPlan.billingPeriod})`;
+
+	const [selectedValue, setSelectedValue] = useState<number | null>(null);
+
+	const [isOtherAmountSelected, setIsOtherAmountSelected] =
+		useState<boolean>(false);
+
+	const [hasInteractedWithOtherAmount, setHasInteractedWithOtherAmount] =
```

Comment:
happy to leave this as is since it's the same logic as another screen - but can't but feel some of this logic could be simplified

Reply:
Agreed - I'll pop a ticket in our health backlog to cover this refactor